### PR TITLE
Refactor iptables module to prepare for nftables implementation

### PR DESCRIPTION
**Pull request by thomasferrandiz**
## Description
This PR creates a trafficmngr package which declares the TrafficManager interface.
This interface holds the currently used chains and rules management functions for the iptables module.

This will allow us to also provide an nftables-based implementation of this interface.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: pkg/trafficmngr/trafficmngr.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	CreateIP4Chain(table, chain string)
+	CreateIP6Chain(table, chain string)
+	MasqRules(cluster_cidrs []ip.IP4Net, lease *lease.Lease) []IPTablesRule
+	MasqIP6Rules(cluster_cidrs []ip.IP6Net, lease *lease.Lease) []IPTablesRule
+	ForwardRules(flannelNetwork string) []IPTablesRule
+	SetupAndEnsureIP4Tables(getRules func() []IPTablesRule, resyncPeriod int)
+	SetupAndEnsureIP6Tables(getRules func() []IPTablesRule, resyncPeriod int)
+	DeleteIP4Tables(rules []IPTablesRule) error
+	DeleteIP6Tables(rules []IPTablesRule) error
+}
```
Comment (manuelbuil) [inline pkg/trafficmngr/trafficmngr.go:39]: I like the change but taking advantage of it, I think we can refactor code a little bit. I think it is possible to simplify this interface and have only:
* ForwardRulesToAccept (which would be called here: https://github.com/flannel-io/flannel/blob/master/main.go#L356-L362 and would hide most of that code)
* AddMasqRules (which would be called here https://github.com/flannel-io/flannel/blob/master/main.go#L340-L390 and would hide most of that code)

Then, the rest of the calls (recycle, which ends up calling Delete; CreateIPChain...) can be private functions inside the implementation of iptables or nftables. That would also make `main.go` a bit cleaner. Right now we are even defining functions https://github.com/flannel-io/flannel/blob/master/main.go#L356-L362 which I find very ugly.

What do you think?

**Review by manuelbuil — CHANGES_REQUESTED** [review]
So there will be a

File: pkg/trafficmngr/trafficmngr.go (hunk 1 above)
Comment (thomasferrandiz) [reply pkg/trafficmngr/trafficmngr.go:39]: I agree
I was planning to do that in another PR to keep things readable :-)

I can do it in one PR if you prefer

Comment (manuelbuil) [reply pkg/trafficmngr/trafficmngr.go:39]: that's also fine for me

**Review by manuelbuil — APPROVED** [review]

**Pull request merged**